### PR TITLE
GitHub Actions: Add Ubuntu workflow

### PR DESCRIPTION
**Pull request by Holzhaus**

### CODE REVIEW

File: .github/workflows/build.yml [hunk 1]
```diff
@@ -0,0 +1,112 @@
...
+      with:
+        cmake-version: '3.13.x'
+    - name: Create build directory
+      run: mkdir cmake_build
+    - name: Prepare ccache timestamp
+      id: ccache_cache_timestamp
+      shell: cmake -P {0}
+      run: |
+        string(TIMESTAMP current_date "%Y-%m-%d-%H;%M;%S" UTC)
+        message("::set-output name=timestamp::${current_date}")
```
Comment (Be-ing) [inline .github/workflows/build.yml:63]: Why is this needed?

File: .github/workflows/build.yml [hunk 2]
```diff
@@ -0,0 +1,112 @@
...
+    - name: Set up ccache cache
+      uses: actions/cache@v1.1.0
+      with:
+        path: .cache/ccache
+        key: ${{ github.job }}-ccache-${{ steps.ccache_cache_timestamp.outputs.timestamp }}
```
Comment (Be-ing) [inline .github/workflows/build.yml:68]:
```suggestion
        key: ${{ github.job }}-ccache-${{ steps.ccache_cache_timestamp.outputs.timestamp }}
```
Putting the timestamp in the cache key is redundant. GitHub Actions already compares caches based on modification time. The cache key should contain an identifying string for the job and the branch. See https://github.com/mixxxdj/mixxx/pull/3305/files#diff-ba2a15f6a0557b1f58421c9e2f89dedd770ebb0ec79eb3be817268246ece1959R36

File: .github/workflows/build.yml [hunk 3]
```diff
@@ -0,0 +1,115 @@
...
+          protobuf-compiler \
+          qt5-default \
+          qt5keychain-dev \
+          qtscript5-dev \
+    - name: Set up cmake
+      uses: jwlawson/actions-setup-cmake@v1.4
+      with:
+        cmake-version: '3.13.x'
+    - name: Create build directory
+      run: mkdir cmake_build
```
Comment (Be-ing) [inline .github/workflows/build.yml:57]: Does this really need to be its own step? Can't you move it to the build step?

File: .github/workflows/build.yml [hunk 4]
```diff
@@ -0,0 +1,115 @@
...
+      shell: cmake -P {0}
+      run: |
+        string(TIMESTAMP current_date "%Y-%m-%d-%H;%M;%S" UTC)
+        message("::set-output name=timestamp::${current_date}")
+    - name: Set up ccache cache
+      uses: actions/cache@v2
+      env:
+        cache-name: ccache
+      with:
+        path: .cache/ccache
```
Comment (Be-ing) [inline .github/workflows/build.yml:69]: This is incorrect.
```suggestion
        path: ~/.ccache
```

Comment (Holzhaus) [reply .github/workflows/build.yml:69]: Are you sure?

> **cache_dir (CCACHE_DIR)**
> This option specifies where ccache will keep its cached compiler outputs. It will only take effect if set in the system-wide configuration file or as an environment variable. **The default is $XDG_CACHE_HOME/ccache if XDG_CACHE_HOME is set, otherwise $HOME/.cache/ccache**. Exception: If the legacy directory $HOME/.ccache exists then that directory is the default.

https://ccache.dev/manual/4.0.html#_configuration_options

Comment (Be-ing) [reply .github/workflows/build.yml:63]: Yes I am sure. I manually reran the build and the build time was the same despite the cache being loaded. I tested this with my branch too. If you run `ccache -p` on GitHub Actions it shows `~/.ccache` as the cache directory.

File: .github/workflows/build.yml [hunk 5]
```diff
@@ -0,0 +1,109 @@
...
+      run: mkdir cmake_build
+    - name: Set up ccache cache
+      uses: actions/cache@v2
+      env:
+        cache-name: ccache
+      with:
+        path: .cache/ccache
+        key: ${{ env.cache-name }}-${{ runner.os }}-${{ github.head_ref }}-${{ github.run_number }}
+        restore-keys: |
+          ${{ env.cache-name }}-${{ runner.os }}-${{ github.head_ref }}
```
Comment (uklotzde) [inline .github/workflows/build.yml:66]: The setup of this action is customized for ccache. The indirection `${{ env.cache-name }}` does not help here. Why not use `ccache` literally here?

According to the [examples](https://github.com/actions/cache/blob/main/examples.md) the preferred or recommended pattern seems to be `${{ runner.os }}-ccache-*`.

File: .github/workflows/build.yml [hunk 6]
```diff
@@ -0,0 +1,109 @@
...
+          libwavpack-dev \
+          portaudio19-dev \
+          protobuf-compiler \
+          qt5-default \
+          qt5keychain-dev \
+          qtscript5-dev \
+    - name: Set up cmake
+      uses: jwlawson/actions-setup-cmake@v1.4
+      with:
+        cmake-version: '3.13.x'
```
Comment (uklotzde) [inline .github/workflows/build.yml:55]: What's the reasoning behind the selected version? Is it the minimum required version or the version included in Ubuntu 18.04. Please add a comment about this implicit dependency that also defines when the version needs to be updated.

Comment (Holzhaus) [reply .github/workflows/build.yml:55]: It's the minimum required version defined in our CMakeLists.txt. That should make sure that we don't use functions that are not present in older CMake versions without updating that value.

Comment (uklotzde) [reply .github/workflows/build.yml:55]: :+1: `# Minimum required version in CMakeLists.txt` ;)

File: .github/workflows/build.yml (hunk 4 above)
Comment (Holzhaus) [reply .github/workflows/build.yml]: Probably because the Ubuntu is old 🤷

File: .github/workflows/build.yml (hunk 5 above)
Comment (Holzhaus) [reply .github/workflows/build.yml]: Fixed.

File: .github/workflows/build.yml (hunk 4 above)
Comment (uklotzde) [reply .github/workflows/build.yml]: Why not add both paths, just in case? One location will always be empty.

Comment (uklotzde) [reply .github/workflows/build.yml]: Fedora 33 with  CCache 3.7.11 also defaults to ~/.ccache. The new CCache 4 location ~/.cache/ccache is ignored and ~/.ccache is re-created implicitly.

Comment (uklotzde) [reply .github/workflows/build.yml]: Version 4.0 has only been released a month ago. Version 3.7 will not disappear soon. By explicitly creating `~/.ccache` if it does not exist we only need to cache a single directory.

Comment (Be-ing) [reply .github/workflows/build.yml]: We build with the oldest Ubuntu version we support so yes we will not be using ccache 4 on Linux for a long time. (I have just added ccache 4 to the macOS build environment.)

**Comment by Be-ing** [comment]
Repeated build with ccache only took 5 minutes :rocket:
